pkgs/development/compilers/edk2/default.nix

```diff
@@ -1,16 +1,25 @@
-{ stdenv, fetchFromGitHub, fetchpatch, libuuid, python2, iasl }:
+{ stdenv, buildPackages, fetchFromGitHub, fetchpatch, libuuid, python2 }:
 
 let
-  pythonEnv = python2.withPackages(ps: [ps.tkinter]);
-
-targetArch = if stdenv.isi686 then
-  "IA32"
-else if stdenv.isx86_64 then
-  "X64"
-else if stdenv.isAarch64 then
-  "AARCH64"
-else
-  throw "Unsupported architecture";
+# Given a stdenv, returns the edk2-valid arch.
+envToArch = env:
+  if env.isi686 then
+    "IA32"
```

Comment:
These keep showing up. We probably should put them in lib/systems/ as ```uefiShortName``` or something to avoid the duplication.

Reply:
They do come often in similar sounding names, but all-caps they're pretty much the only instances :/. These ones, though, seem to be really specific to their build system.

`iso-image.nix` has those, which are not entirely equivalent, even ignoring case.

```
363   # Name used by UEFI for architectures.
364   targetArch =
365     if pkgs.stdenv.isi686 then
366       "ia32"
367     else if pkgs.stdenv.isx86_64 then
368       "x64"
369     else if pkgs.stdenv.isAarch64 then
370       "aa64"
371     else
372       throw "Unsupported architecture";
```